### DISCUSSION TITLE
Cannot paste text from Microsoft Word .docx file (Windows 11)

### DISCUSSION THREAD

**Discussion by StephenHoust** [danny-avila/LibreChat#2662] [Troubleshooting]
### What happened?

Nothing happens (no text is pasted in the message box).

### Steps to Reproduce

1. Open .docx file in Windows 11 using latest version of Microsoft Word.
2. Copy text from file.
3. Attempt to paste into LibreChat message dialog box.

### Attempts to resolve:
1. Update LibreChat (via git pull & docker compose pull). Have run this step approximately 6 times since issue started.
2. Forced refresh of cache in Firefox (via CTRL+F5 countless times).
3. Restarting both computers.

To note, this is observed when running LibreChat hosted on a Linux desktop PC (OpenSUSE Tumbleweed, fully updated) and connecting to that instance via LAN on a Windows 11 desktop PC. I have not attempted any other setups. This was _not_ an issue roughly two weeks ago. I saw the other related pasting issues last week, so I'm not sure why the resolution to those issues did not seem to resolve this for me as well, as it would seem to be the exact same issue?

### What browsers are you seeing the problem on?

Firefox

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I'm also on windows 11, and can confirm this is resolved on the latest builds: 

https://github.com/danny-avila/LibreChat/assets/110412045/b767e1c0-d3d6-42c8-ab35-60767e05f9b8

How are you updating?

What is the output when you run `git log` ?

**Reply by StephenHoust** [reply]
I'm updating via command line on the Linux computer running the LibreChat instance. See attached git log.

[git_log.txt](https://github.com/danny-avila/LibreChat/files/15280724/git_log.txt)

**Comment by danny-avila** [comment]
From the recommend steps, I would add `git fetch origin` before this:

```bash
# Stop the running container(s)
docker compose down

# Pull latest project changes
git pull

# Pull the latest LibreChat image (default setup)
docker compose pull

# If building the LibreChat image Locally, build without cache (legacy setup)
# docker compose build --no-cache

# Start LibreChat
docker compose up
```
https://docs.librechat.ai/install/installation/docker_compose_install.html#updating-librechat

**Reply by StephenHoust** [reply]
I've been following these steps, without the git fetch origin. I ran that at the beginning of the series just now (and, to be clear, I am not running the commented out docker compose build --no-cache line). The git log I pasted in the previous comment was right after running these commands again with git fetch origin first, as recommended. Still no success on pasting from the Windows 11 device.

Of interest, I cannot replicate this issue from my Linux-based desktop that is hosting the LibreChat instance. I've tried both LibreOffice (best substitution I can do for Word) and PyCharm Professional, since someone had a previous issue about pasting from PyCharm, with no problems.

That leads me to believe that most likely it is some unusual issue where the hosting device can paste properly, but not other network devices? Or just something wonky happened with settings somewhere that is requiring a clean reinstall? Although the fact that it works fine on one PC, but not another is a bit of a head scratcher.

**Reply by danny-avila** [reply]
@StephenHoust sometimes, it's the browser cache to blame as the browser may be pulling the same static files from a version ago. Resetting cookies/cache/localStorage/using a different browser may help.

Sorry you're running into this. I've had many frustrations with the Browser text input API but finally jumped over a big hurdle development-wise that will eliminate this issue altogether soon. Unfortunately, I did not realize that rich text compatibility had to be re-implemented in the process.

**Reply by StephenHoust** [reply]
I had reset those and done a full clear of cookies and cache. I was planning on trying a different browser, but had not gotten a chance to do so yet. That said, while it wasn't working yesterday still, for whatever reason after pulling the git and docker updates this morning I'm able to paste now without issue. Not sure if an update fixed my problem or if it was on my side but somehow magically decided to have a clean cache today.

Either way, it is resolved now. Appreciate the assistance!

**Reply by danny-avila** [reply]
So glad it’s working for you and I can assure you that a regression here will now be a thing of the past :)
